### PR TITLE
Various hint improvements

### PR DESCRIPTION
**Pull request by glacambre**
This adds a rapid version for each possible hint action and also makes hint mode less janky (thus fixing https://github.com/cmcaine/tridactyl/issues/848).

Since this brings large changes to hint mode, https://github.com/cmcaine/tridactyl/issues/854, https://github.com/cmcaine/tridactyl/issues/745 and https://github.com/cmcaine/tridactyl/issues/449 might accidentally be fixed and would need to be revisited once this PR reaches stable.

The most important commit of this PR is 6c3c04f, reading its commit message might help you understand the whole PR. Sorry it got so big. I made sure to test every hint action and everything seems to work, please let me know if I missed a bug.

### CODE REVIEW

**Comment by bovine3dom** [comment]
I'm very amused that you've dedeprecated `-W`.

This might take me a while to review.

**Comment by bovine3dom** [comment]
Huh. I could have sworn that I'd merged this already...

I'll try to find time to review it tomorrow.

**Comment by glacambre** [comment]
Thank you for noticing this. I swear I tested this for every other mode, I just forgot `hint` alone :<

Also, I just realized that might be useful to have the result of hinting be an array. Hinting and leaving hint mode without having selected any element would return an empty array, normal hinting would return an array with a single element and rapid hinting would return an array of multiple elements.

**Comment by glacambre** [comment]
Actually now that I've implemented always returning arrays I think it'd be better to have normal hinting return the element rather than an array made of a single elements. It would make piping easier I think.

File: src/excmds.ts [hunk 1]
```diff
@@ -2880,13 +2880,12 @@ import * as hinting from "./hinting"
         - -c [selector] hint links that match the css selector
           - `bind ;c hint -c [class*="expand"],[class="togg"]` works particularly well on reddit and HN
         - -w open in new window
-            -wp open in new private window
-        - `-pipe selector key` e.g, `-pipe * href` returns the key. Only makes sense with `composite`, e.g, `composite hint -pipe * textContent | yank`.
-        - `-W excmd...` append hint href to excmd and execute, e.g, `hint -W exclaim mpv` to open YouTube videos. Use `composite hint -pipe | [excmd]` instead.
-        - -q* quick (or rapid) hints mode. Stay in hint mode until you press <Esc>, e.g. `:hint -qb` to open multiple hints in the background or `:hint -qW excmd` to execute excmd once for each hint
+        - -wp open in new private window
+        - `-pipe selector key` e.g, `-pipe * href` returns the key. Only makes sense with `composite`, e.g, `composite hint -pipe * textContent | yank`. If you don't select a hint (i.e. press <Esc>), will return undefined.
```
Comment (bovine3dom) [inline src/excmds.ts]: I think I'd rather return an empty string / abort somehow, as coercion to text will literally put "undefined" in the clipboard, etc.

e.g. `hint -pipe | tabopen` shouldn't open a blank tab when you hit escape.

Comment (glacambre) [reply src/excmds.ts]: We currently can't abort piped commands unless we throw an exception and these will be caught and displayed in the command line. We probably don't want to do that everytime the user just leaves hint mode without having selected anything, so I made it return an empty string.

**Pull request merged**